2dfft_utils/analysis/spiral_overlay.py

```diff
@@ -0,0 +1,276 @@
+                        All necessary libraries included in Ureka package.
+    
+                        Eqns used:
+    
+                            r = ae^(b*theta)
+    
+                            x = r*cos(theta)
+                            y = r*sin(theta)
+    
+                            tan(phi) = b
+                            phi = pitch angle
+    
+                        Edit theta_max, title, origin of the plot (imshow)
+                        & the colormap (imshow(...,cmap=..) as needed.
+                        Colormaps schemes available can be seen at:
+                        http://matplotlib.org/examples/color/colormaps_reference.html
+    
+                        Recommended cmap options are included as comments.
+                        
+                        It is recommended that you use cropped images with this program. There
+                        is a marked decrease in performance as image size increases.
+                    
+                        Rotation angle and pitch can be changed while the program is running
+                        using the sliders below the plot. The reset button initializes the
+                        slider to default values (0.0 for rotation, 25.0 for pitch).
+                        
+                        Scale can be swapped between logarithmic and linear modes with the radio
+                        buttons on the left side of the plot. Defaults to linear scale.
+                        
+                        Chirality (corresponding to the sign of the pitch angle: CCW = +, CW = -)
+                        can be changed with the second set of radio buttons on the left side of
+                        the plot. Defaults to CCW.
+                        
+                        Arm number (corresponding to mode) can be changed with the radio buttons
+                        on the right side of the plot. Defaults to 2 arms.
+'''
+
+#-----------------------------Program begins----------------------------
+#!/usr/bin/python
+from pylab import *
+import numpy as np
+import glob
+from astropy.io import fits
+from matplotlib.colors import LogNorm
+import matplotlib.pyplot as plt
+from matplotlib.widgets import Slider, Button, RadioButtons
+import sys
+
+#-----------------------------Default/initial values-----------------------------
+DEFAULT_ARMS = 2    #Between 1 and 6, inclusive
+DEFAULT_PITCH = 23.0  #Between about -89.12 and 89.12 Anything bigger causes overflow issues.
+DEFAULT_ROTATION = 0.0 #Between -180.0 and 180.0, inclusive
+DEFAULT_COLORSCALE = 'n' #n is linear, yg is logarithmic grey scale, ys is logarithmic seismic scale
+#-----------------------------Command line input-----------------------------
+if len(sys.argv) != 1:
+    if sys.argv[1][-4:] == ".fit" or sys.argv[1][-5:] == ".fits":
+        gal_file = sys.argv[1]
+    else:
+        print "Usage: python spiral_overlay.py <filename>"
```

Comment:
There's a more Pythonic way to output error messages.  

Once you start doing things like write unit tests, if you get failing behavior, those tests won't necessarily catch it if you just use print statements. 

See [here](https://github.com/astropy/astroplan/blob/master/astroplan/plots/sky.py#L127) for an example (although note that this example doesn't necessitate exiting the program).